src/core/eme/eme_manager.ts

```diff
@@ -170,7 +177,36 @@ export default function EMEManager(
                               value: initializationData });
       }
 
-      if (!handledInitData.storeIfNone(initializationData, true)) {
+      const lastKeyUpdate$ = new ReplaySubject<IKeyUpdateValue>(1);
+
+      // First, check that this initialization data is not already handled
+      if (options.singleLicensePer === "content" && handledSessions.getLength() > 0) {
```

Comment:
Shouldn't we have two different objects ? `handledSessions` for `singleLicensePer === "init-data"` and another one for "content" ?

Reply:
Wouldn't that just complexify the code?

Also, this code could be adaptable to a `singleLicensePer: "period"` by adding some metadata to both `handledSessions` and the `initializationData` sent to the `EMEManager`

Reply:
It was just a little bit strange for me that the way to check if we already handled the init data was to see if there was at least one handled session.

Reply:
I understand.

The fact is `singleLicensePer: "content"` also means a single MediaKeySession per content. Maybe I should define a `isEmpty` methode on `handledSessions` to make that more readable, and maybe I should find a better name that `handledSessions`, I probably should add a comment too.

Instead of `handledSessions` maybe , `contentSessions`?

This would make:
```js
if (options.singleLicensePer === "content" && !contentSessions.isEmpty()) {
   // If only a single license is necessary per content, we only need to create
  // a single MediaKeySession for that content.
```

Reply:
Yes, I had thought of a new method, without knowing exactly what it could be!

Reply:
Thus, will you add a new method ?

Reply:
Yes! Didn't take the time yet :/

Reply:
Done